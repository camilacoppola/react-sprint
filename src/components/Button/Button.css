@import url("../../generalCSS/colors.css");

.Button-button1{
    border-radius: 3rem;
    border: solid var(--grisT2) 1px;
    padding: 0.3rem 0.5rem;
    max-height: 2rem;
    color: white;
}

.Button-button1:hover{
    background-color: var(--colorVerde);
}

.Button-link1{
    border-radius: 3rem;
    border: solid var(--grisT2) 1px;
    padding: 0.3rem 0.5rem;
    max-height: 2rem;
    background-color: var(--grisT2);
    color: white;
}

.Button-link1:hover{
    background-color: var(--grisT2Selected);
}