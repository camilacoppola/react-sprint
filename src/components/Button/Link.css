.Button-link1{
    margin-right: 5px;
    border-radius: 15px;
    border: none;
    background-color: #494f51;
    color: white;
    cursor: pointer;
    padding: 8px 10px;
    font-size: 12px;
}