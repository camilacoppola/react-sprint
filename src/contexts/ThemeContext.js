import React, { createContext, useState } from 'react';

export const consumeThemeContext = createContext();

const ThemeContext = ({children}) => {

    let modoOscuro = localStorage.getItem('modoOscuro');
    if (modoOscuro === undefined || modoOscuro === null){
        localStorage.setItem('modoOscuro', false);
        modoOscuro = false;
    }

    const [darkModeTheme, setTheme] = useState(modoOscuro);

    const toggleTheme = () => {
        setTheme(!darkModeTheme);
    }

    return (
        <consumeThemeContext.Provider value={{darkModeTheme, toggleTheme}}>
            {children}
        </consumeThemeContext.Provider>
    )
}

export default ThemeContext;