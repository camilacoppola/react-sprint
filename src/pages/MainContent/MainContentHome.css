body{
    background-color: black;
}
.Home-contenedor_home{
    display: flex;
    flex-direction: column;
    justify-content: space-between;
   
}

.Home-productos_home, .Home-tiendas_home{
    display: flex;
    justify-content: space-between;
    background-color: #222222;
    border-radius: 1rem;
    padding: 0.5rem 1rem;
    box-shadow: 0px 4px 4px 0px #00000040;
    margin: 1rem;
    color: white;
}
.Home-contenedorButton_home, .Home-imgDescripcion_home{
    display: flex;
    align-items: center;
}

.Home-button_home {
    margin-right: 5px;
    border-radius: 15px;
    border: none;
    padding: 0.5rem;
    background-color: #494f51;
    color: white;
    cursor: pointer;
}

.Home-logoProductsTienda_home{
    filter: invert(100%);
    margin-right: 0.5rem;
}