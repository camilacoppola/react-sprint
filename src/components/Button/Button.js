import React from 'react'
import './Button.css'
import './Link.css'
import { Link } from 'react-router-dom';

const Button = ({tipoButton, textoDescripcion, linkTo, claseBoton}) => {
    let botonAElegir;
    switch(tipoButton){
        case "tipoBoton1": 
            botonAElegir = (<button className= 'Button-button1'>{textoDescripcion}</button>)
            break;
        case "tipoBoton2":
            botonAElegir = (<button className= 'Button-button2'>{textoDescripcion}</button>)
            break;
        case "tipoLink1": 
            botonAElegir = <Link to={linkTo} className={'Button-link1 ' + claseBoton}>{textoDescripcion}</Link>
            break;
        case "tipoLink2":
            botonAElegir = <Link to={linkTo} className={'Button-link2 ' + claseBoton}>{textoDescripcion}</Link>
            break;
        default : 
            throw 'Tipo de boton incorrecto'
      }
    

  return (
   botonAElegir
)

}

export default Button