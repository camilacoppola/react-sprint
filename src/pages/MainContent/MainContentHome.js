import React, { useContext } from 'react';
import './MainContentHome.css'
import logoProducts from '../../assets/img/package-variant-closed.svg'
import logoTiendas from '../../assets/img/store.svg'
import Button from '../../components/Button/Button';
import { consumeThemeContext } from '../../contexts/ThemeContext';

const MainContentHome = () => {
    const {darkModeTheme} = useContext(consumeThemeContext)

  return (
    <div className={darkModeTheme? 'Home-contenedor_home': 'Home-contenedor_home Home-modo_claro'}>
        <div className='Home-productos_home'>
            <div className='Home-imgDescripcion_home'>
                <img className='Home-logoProductsTienda_home'src={logoProducts} alt="Logo productos" />
                <p> 123 Productos</p>
            </div>
            <div className='Home-contenedorButton_home'>
                <Button tipoButton={"tipoLink1"} textoDescripcion={"Ver Listado"} linkTo={'/products'} />
                <Button tipoButton={"tipoLink1"} textoDescripcion={"Agregar Producto"} linkTo={'/products/new'} />
            </div>
        </div>
        <div className='Home-tiendas_home'>
            <div className='Home-imgDescripcion_home'>
                <img className='Home-logoProductsTienda_home'src={logoTiendas} alt="Logo productos" />
                <p>10 tiendas</p>
            </div>
            <div className='Home-contenedorButton_home'>
                <Button tipoButton={"tipoLink1"} textoDescripcion={"Ver Listado"} linkTo={'/stores'} />
                <Button tipoButton={"tipoLink1"} textoDescripcion={"Agregar Tienda"} linkTo={'/stores/new'} />
            </div>
        </div>
    </div>
  )
}

export default MainContentHome