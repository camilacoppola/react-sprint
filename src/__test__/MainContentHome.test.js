import { fireEvent, render,screen} from "@testing-library/react";
import MainContentHome from "../pages/MainContent/MainContentHome";
import ThemeContext  from '../contexts/ThemeContext';
import {MemoryRouter} from 'react-router-dom';
import '@testing-library/jest-dom';
import Button from "../components/Button/Button";
import userEvent from "@testing-library/user-event";


describe('MainContentHome', ()=>{

    beforeEach(()=>{render(
        <ThemeContext>
            <MainContentHome/>
        </ThemeContext>,
        {wrapper:MemoryRouter}
    )})

    test('Verificar titulo de productos', ()=>{
        const tituloProductos = screen.getByText(/123 Productos/i);
        expect(tituloProductos).toBeInTheDocument();
    })

    test('Verificar titulo de tiendas', () =>{
        const tituloTiendas = screen.getByText(/10 Tiendas/i);
        expect(tituloTiendas).toBeInTheDocument()
    })

    test('Funcionamiento del boton - Ver listado',()=>{
        const link = screen.getAllByRole("link")
        userEvent.click(link[0])
        expect(link[0]).toMatchSnapshot()
        expect(link[0].getAttribute("href")).toMatch('/products')
    })


    
})