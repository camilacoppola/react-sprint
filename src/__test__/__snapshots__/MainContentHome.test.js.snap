// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContentHome Funcionamiento del boton - Ver listado 1`] = `
<a
  class="Button-link1 "
  href="/products"
>
  Ver Listado
</a>
`;
